Source/WebInspectorUI/UserInterface/Views/OverrideDeviceSettingsPopover.js

```diff
@@ -0,0 +1,324 @@
+            ],
+        };
+
+        let mediaSettingsGroup = {
+            name: WI.UIString("Media"),
+            settings: [
+                {name: WI.UIString("Allow media capture on insecure sites"), setting: InspectorBackend.Enum.Page.Setting.MediaCaptureRequiresSecureConnection, value: false},
+                {name: WI.UIString("Disable ICE candidate restrictions"), setting: InspectorBackend.Enum.Page.Setting.ICECandidateFilteringEnabled, value: false},
+                {name: WI.UIString("Use mock capture devices"), setting: InspectorBackend.Enum.Page.Setting.MockCaptureDevicesEnabled, value: true},
+
+                // FIXME: Find out whether we need to keep this.
+                // {name: WI.UIString("Disable Encryption"), setting: InspectorBackend.Enum.Page.Setting.WebRTCEncryptionEnabled, value: false},
+            ],
+        };
+
+        const settingsGroups = [coreSettingsGroup];
+
+        if (InspectorFrontendHost.isRemote) {
+            settingsGroups.push(compatibilitySettingsGroup);
+            settingsGroups.push(securitySettingsGroup);
+            settingsGroups.push(privacySettingsGroup);
+        }
+
+        settingsGroups.push(mediaSettingsGroup);
+
+        let fragment = document.createDocumentFragment();
+
+        for (let group of settingsGroups) {
+            let settingsGroupHeadingElement = fragment.appendChild(document.createElement("h1"));
+            settingsGroupHeadingElement.textContent = group.name;
+
+            for (let {name, setting, value} of group.settings) {
+                if (!setting)
+                    continue;
+
+                let labelElement = fragment.appendChild(document.createElement("label"));
+
+                let checkboxElement = labelElement.appendChild(document.createElement("input"));
+                checkboxElement.type = "checkbox";
+                checkboxElement.checked = WI.deviceSettingsManager.overridenDeviceSettings.has(setting);
+                checkboxElement.addEventListener("change", (event) => {
+                    WI.deviceSettingsManager.overrideDeviceSetting(setting, value, (enabled) => {
+                        checkboxElement.checked = enabled;
+                    });
+                });
+
+                labelElement.append(name);
+            }
+        }
+
+        return fragment;
+    }
+
+    _createUserAgentSection()
+    {
+        let fragment = document.createDocumentFragment();
+
+        let userAgentTitle = fragment.appendChild(document.createElement("h1"));
+        userAgentTitle.textContent = WI.UIString("User Agent");
+
+        let userAgentValue = fragment.appendChild(document.createElement("div"));
+        userAgentValue.classList.add("select-with-input-container");
+
+        let userAgentValueSelect = userAgentValue.appendChild(document.createElement("select"));
+
+        const userAgents = [
```

Comment:
Our patches have crossed in the wind, and this is no longer quite correct. It needs to check the WebKitAdditions global object for user agents, and fall back to these. Double check the current ToT implementation in Main.js

Reply:
Good catch! I brought over the `WebKitAdditions` check.